scripts/test-infra/test-e2e-presubmit.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/sh
+set -o errexit
+
+# Configure environment
+export IMAGE_REGISTRY="localhost:5001"
+export IMAGE_TAG_NAME=$(git describe --tags --dirty --always)
+export MANIFESTS_DIR="deployment/overlays/default"
+export KUBECONFIG="/tmp/$IMAGE_TAG_NAME"
+
+# create registry container unless it already exists
+reg_name='kind-registry'
+reg_port='5001'
+if [ "$(docker inspect -f '{{.State.Running}}' "${reg_name}" 2>/dev/null || true)" != 'true' ]; then
+  docker run \
+    -d --restart=always -p "127.0.0.1:${reg_port}:5000" --name "${reg_name}" \
+    registry:2
+fi
+
+# create a cluster with the local registry enabled in containerd
+cat <<EOF | kind create cluster --kubeconfig $KUBECONFIG --name $IMAGE_TAG_NAME --config=-
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+- role: control-plane
+- role: worker
+- role: worker
+containerdConfigPatches:
+- |-
+  [plugins."io.containerd.grpc.v1.cri".registry.mirrors."localhost:${reg_port}"]
+    endpoint = ["http://${reg_name}:5000"]
+EOF
+
+# connect the registry to the cluster network if not already connected
+if [ "$(docker inspect -f='{{json .NetworkSettings.Networks.kind}}' "${reg_name}")" = 'null' ]; then
+  docker network connect "kind" "${reg_name}"
+fi
+
+cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: local-registry-hosting
+  namespace: kube-public
+data:
+  localRegistryHosting.v1: |
+    host: "localhost:${reg_port}"
+    help: "https://kind.sigs.k8s.io/docs/user/local-registry/"
+EOF
+
+# build local image
+make image \
+  IMAGE_REGISTRY=$IMAGE_REGISTRY \
+  IMAGE_TAG_NAME=$IMAGE_TAG_NAME \
+  IMAGE_NAME=$IMAGE_NAME \
+  IMAGE_EXTRA_TAGS=$IMAGE_EXTRA_TAGS
```

Comment:
We take the neededs vars from env 

```suggestion
make image
```